backends/ebpf/tests/ptf/common.py

```diff
@@ -77,8 +87,23 @@ def setUp(self):
         head, tail = os.path.split(self.p4_file_path)
         filename = tail.split(".")[0]
         self.test_prog_image = os.path.join("ptf_out", filename + ".o")
+        logger.info("\nUsing test params: %s", testutils.test_params_get())
+        if "namespace" in testutils.test_params_get():
+            self.switch_ns = testutils.test_param_get("namespace")
+        self.interfaces = testutils.test_param_get("interfaces").split(",")
+
+        # fetch data plane ports from network namespace
+        global DP_PORTS
+        next_idx = 0
+        for intf in self.interfaces:
+            if intf != "psa_recirc":
+                DP_PORTS[next_idx] = self.get_dataplane_port_number(intf)
+                next_idx += 1
 
         p4args = "--Wdisable=unused --max-ternary-masks 3"
+        for idx, dp_port in DP_PORTS.items():
+            p4args += " -DPORT{}={}".format(idx, dp_port)
+        p4args += " -DPSA_RECIRC={}".format(self.get_dataplane_port_number("psa_recirc"))
```

Comment:
When these macros are defined like `((PortId_t) dp_port)` then we would omit the type casting in the P4 programs...

```suggestion
            p4args += " -DPORT{}=\"((PortId_t) {})\"".format(idx, dp_port)
        p4args += " -DPSA_RECIRC="((PortId_t) {})\"".format(self.get_dataplane_port_number("psa_recirc"))
```

Reply:
I don't see much value in doing that. In fact, casting to `PortId_t` is a standard part of any PSA program.. I'd keep it as is.